apps/linker/src/app/components/show-result/show-result.component.ts

```diff
@@ -0,0 +1,133 @@
+import { OAuthService, OAuthStorage } from 'angular-oauth2-oidc';
+import { MatDialog } from '@angular/material/dialog';
+import {
+  AttributesManagerService,
+  AuthzResolverService,
+  RTMessagesManagerService,
+  User,
+} from '@perun-web-apps/perun/openapi';
+import { StoreService } from '@perun-web-apps/perun/services';
+import { UserFullNamePipe } from '@perun-web-apps/perun/pipes';
+import { parseQueryParams } from '@perun-web-apps/perun/utils';
+
+// eslint-disable-next-line
+declare let require: any;
+
+@Component({
+  selector: 'perun-web-apps-show-result',
+  templateUrl: './show-result.component.html',
+  styleUrls: ['./show-result.component.css'],
+})
+export class ShowResultComponent implements OnInit {
+  linkerResult: LinkerResult = 'UNKNOWN_ERROR';
+  loading = false;
+
+  constructor(
+    private route: ActivatedRoute,
+    private dialog: MatDialog,
+    private oauthService: OAuthService,
+    private attributeService: AttributesManagerService,
+    private storeService: StoreService,
+    private authzService: AuthzResolverService,
+    private rtMessages: RTMessagesManagerService,
+    private userFullNamePipe: UserFullNamePipe,
+    private oAuthStorage: OAuthStorage
+  ) {}
+
+  ngOnInit(): void {
+    this.route.params.subscribe((params) => {
+      this.linkerResult = params['result'] as LinkerResult;
+      (window.opener as Window).postMessage(this.linkerResult, '*');
+    });
+  }
+
+  closeWindow(): void {
+    window.close();
+  }
+
+  loginAgain(): void {
+    let queryParams = location.search.substring(1);
+    let index = 1;
+    if (this.linkerResult === 'IDENTITY_UNKNOWN') {
+      for (let i = 1; i < 10; i++) {
+        const formAccessToken = parseQueryParams('formerToken' + i.toString(), queryParams);
+        if (formAccessToken === undefined || formAccessToken == null || formAccessToken === '') {
+          index = i;
+          break;
+        }
+      }
+      queryParams =
+        queryParams + '&formerToken' + index.toString() + '=' + this.oauthService.getAccessToken();
+    }
+    sessionStorage.setItem('auth:queryParams', queryParams);
+    this.oauthService.initLoginFlow();
+  }
+
+  sendMessageToSupport(): void {
+    this.loading = true;
+    const currentUser = this.storeService.getPerunPrincipal().user;
+
+    const queryParams = location.search.substr(1);
+    this.oAuthStorage.setItem('access_token', parseQueryParams('formerToken', queryParams));
+    this.authzService.getPerunPrincipal().subscribe((principal) => {
+      const formerUser = principal.user;
+      this.rtMessages
+        .sentMessageToRTWithQueue(
+          'perun',
+          'Account linking: The accounts could not be automatically linked.',
+          this.getFullEmailBody(currentUser, formerUser)
+        )
+        .subscribe(
+          () => {
+            this.linkerResult = 'MESSAGE_SENT_TO_SUPPORT';
+            (window.opener as Window).postMessage(this.linkerResult, '*');
```

Comment:
There are multiple instances of accessing `window.opener` and casting it to `Window`, is it possible to save this to a variable and prevent this need to cast repeatedly.